lib/internal/modules/esm/loader.js

```diff
@@ -89,10 +136,20 @@ class DefaultModuleLoader {
    */
   loaderType = 'default';
 
-  constructor() {
+  /**
+   * Loader to pass requests to.
+   */
+  #delegate = null;
+
+  constructor(delegate = null) {
     if (getOptionValue('--experimental-network-imports')) {
       emitExperimentalWarning('Network Imports');
     }
+    this.#delegate = delegate;
+  }
+
+  setDelegate(delegate) {
```

Comment:
Kind of clunky and unsafe… but no worse than https://github.com/nodejs/node/pull/48439/files#diff-72e781a50541066b01a07699fce4956ab4d808d77588bd9c61d4784549664714R24

Reply:
Open to suggestions! 😄

Reply:
Actually, let's address in a follow-up. Not important enough to delay this PR 🙂

But I think a more specific name for "delegate" should happen now because the current is too vague.